### PR TITLE
🔨 Cotisations contrats loi Madelin retraite et mutuelle

### LINKED ISSUE #533: Ajouter les contrat type "Loi madelin" pour les indépendants
Les cotisations à un contrat privé de type loi madelin sont exonérée d'impôt sur le revenu.

https://www.impots.gouv.fr/portail/particulier/questions/je-cotise-un-contrat-madelin-quel-est-mon-avantage-fiscal

### PR DESCRIPTION
**Pull request by lajarre**
Replaces https://github.com/betagouv/mon-entreprise/pull/942

Closes #533

### CODE REVIEW

File: source/components/simulationConfigs/indépendant.yaml [hunk 1]
```diff
@@ -28,6 +28,7 @@ questions:
     - entreprise . ZFU
     - dirigeant . indépendant . cotisations et contributions . exonérations . âge
     - dirigeant . indépendant . cotisations et contributions . exonérations . invalidité
+    - dirigeant . indépendant . cotisations et contributions . cotisations . contrat madelin
```
Comment (mquandalle) [inline source/components/simulationConfigs/indépendant.yaml:31]: C'est non prioritaire les questions Madelin ? Pour moi ça concerne beaucoup d'indépendants (quasi-tous pour la partie mutuelle, et beaucoup aussi pour la retraite supplémentaire), et en tout cas beaucoup plus que pour les autres questions non prioritaire (ZFU, âge, etc.)

**Comment by lajarre** [comment]
> Puis exécuter `yarn run test-regressions` et commiter le résultat

@mquandalle J'ai écrit des tests, mais j'ai qd même pris le temps de réfléchir aux cas de figure "edge case" et vérifier un peu à la mano les résultats pour voir si c'est cohérent sur le plan de la logique avec la documentation. J'ai mis pas mal de commentaires, car ce genre de tests sont assez muets.

File: source/rules/dirigeant.yaml [hunk 2]
```diff
@@ -659,6 +666,111 @@ dirigeant . indépendant . cotisations et contributions . cotisations . déducti
...
+dirigeant . indépendant . contrats madelin:
+  titre: Contrats Madelin
+
+dirigeant . indépendant . contrats madelin . montant:
+  titre: Somme des cotisations à contrats Madelin
+  formule:
+    somme:
+      - mutuelle . montant
+      - retraite . montant
+  contrôles:
```
Comment (lajarre) [inline source/rules/dirigeant.yaml:678]: @johangirod ça te paraît bien comme ça?

File: source/components/simulationConfigs/indépendant.yaml (hunk 1 above)
Comment (lajarre) [reply source/components/simulationConfigs/indépendant.yaml:31]: C'est un déchet je l'ai viré. En fait je n'ai rien à mettre dans cette config, si?

File: test/regressions/simulations-rémunération-dirigeant.yaml [hunk 3]
```diff
@@ -40,3 +40,35 @@ activités:
...
     entreprise . catégorie d'activité . service ou vente: service
     entreprise . catégorie d'activité . restauration ou hébergement: oui
+
+Contrats Madelin:
+  # Cas retraite: la cotisation Madelin est inferieure au plafond => le revenu net de
+  # cotisations (résultat comptable) n'est pas affecté car l'assiette des
+  # cotisations ne change pas:
+  - dirigeant . rémunération totale: 30000
+    entreprise . charges: 10000
+    dirigeant . indépendant . contrats madelin . mutuelle . montant: 10% * plafond sécurité sociale temps plein - 100 # plafond: 10% PSS donc environ 4100
```
Comment (mquandalle) [inline test/regressions/simulations-rémunération-dirigeant.yaml:50]: Ça ne fonctionne pas encore les expressions dans un object "situation" (@johangirod travaille dessus).

File: source/components/simulationConfigs/indépendant.yaml (hunk 1 above)
Comment (johangirod) [reply source/components/simulationConfigs/indépendant.yaml]: A priori non

File: test/regressions/simulations-rémunération-dirigeant.yaml (hunk 3 above)
Comment (johangirod) [reply test/regressions/simulations-rémunération-dirigeant.yaml]: C'est pour bientôt :)

File: source/rules/dirigeant.yaml (hunk 2 above)
Comment (johangirod) [reply source/rules/dirigeant.yaml:678]: C'est top ! Bien vu :+1:
